rskj-core/src/main/java/co/rsk/trie/Trie.java

```diff
@@ -1212,17 +1220,23 @@ public IterationElement next() {
             IterationElement visitingElement = visiting.pop();
             Trie node = visitingElement.getNode();
             TrieKeySlice nodeKey = visitingElement.getNodeKey();
-            // need to visit the left subtree first, then the right since a stack is a LIFO, push the right subtree first,
-            // then the left
-            Trie rightNode = node.retrieveNode((byte) 0x01);
-            if (rightNode != null) {
-                TrieKeySlice rightNodeKey = nodeKey.rebuildSharedPath((byte) 0x01, rightNode.getSharedPath());
-                visiting.push(new IterationElement(rightNodeKey, rightNode));
-            }
-            Trie leftNode = node.retrieveNode((byte) 0x00);
-            if (leftNode != null) {
-                TrieKeySlice leftNodeKey = nodeKey.rebuildSharedPath((byte) 0x00, leftNode.getSharedPath());
-                visiting.push(new IterationElement(leftNodeKey, leftNode));
+
+            int nodeKeyLength = nodeKey.length();
+            // If we're stoping at accounts, do not add children
+            if ((!stopAtAccountDepth) || (nodeKeyLength < (1 + TrieKeyMapper.SECURE_KEY_SIZE + RskAddress.LENGTH_IN_BYTES) * Byte.SIZE))
```

Comment:
you can add some comments to prevent using the iterator in a wrong way

Reply:
done!

Reply:
Done!